src/pip/_internal/network/lazy_wheel.py

```diff
@@ -64,13 +66,30 @@ def __init__(
         self._session, self._url, self._chunk_size = session, url, chunk_size
 
         # initial range request for the end of the file
-        tail = self._stream_response(start="", end=CONTENT_CHUNK_SIZE)
+        try:
+            tail = self._stream_response(start="", end=CONTENT_CHUNK_SIZE)
+        except HTTPError as e:
+            if e.response.status_code != 416:
+                raise
+
+            # The 416 response message contains a Content-Range indicating an
+            # unsatisfied range (that is a '*') followed by a '/' and the current
+            # length of the resource. E.g. Content-Range: bytes */12777
+            content_length = int(e.response.headers['content-range'].rsplit('/', 1)[-1])
```

Comment:
Do we need to add some error handling here in case the server is sending something invalid?